### DISCUSSION TITLE
Dashboards links don't send referer header

### DISCUSSION THREAD

**Discussion by muutech** [grafana/grafana#38266] [Dashboards, Panels & Visualizations]
**What happened**:

We have some dashboard links processing "referer" (for instance for automate creation of ticket and getting automatically the origin url of the problem):

In 7.5.4:

![image](https://user-images.githubusercontent.com/30987258/116451712-21f41000-a85d-11eb-9940-b5d5475b2f69.png)

Click on the link:

![image](https://user-images.githubusercontent.com/30987258/116451776-37693a00-a85d-11eb-93bf-64aea459d084.png)

If we "inspect" the button:

![image](https://user-images.githubusercontent.com/30987258/116451944-6ed7e680-a85d-11eb-97fe-5ab906d49e77.png)

We found rel="noreferrer", causing the referrer not being sent.

**What you expected to happen**:

Include referer header, like previous versions

In previous versions of grafana:

![image](https://user-images.githubusercontent.com/30987258/116448965-f6bbf180-a859-11eb-8253-bc49bf82bd73.png)

Result:

![image](https://user-images.githubusercontent.com/30987258/116449098-18b57400-a85a-11eb-9c5b-193a41415b1e.png)

**How to reproduce it (as minimally and precisely as possible)**:

Add a dashboard link, click on it, check referer is being sent (see "what  happened")

**Anything else we need to know?**:

We discover in this PR: https://github.com/grafana/grafana/pull/31408 this "rel="noreferrer"" is added to all dashboards links, telling the browser to not pass referer, breaking the functionality. The reason, after asking @zoltanbedi  appear to be a security recommendation: 

Quoting from mdn docs:

> Using target="_blank" without rel="noreferrer" and rel="noopener" makes the website vulnerable to window.opener API exploitation attacks (vulnerability description), although note that, in newer browser versions setting target="_blank" implicitly provides the same protection as setting rel="noopener". See browser compatibility for details.

In terms of security, "noreferrer" is needed only for FF versions not supporting "noopener", and it is only vulnerable when target="_blank" is used. Currently all the browsers supported by Grafana implement the implicit setting of rel="noopener" (so, imho is not needed at all, see https://sites.google.com/site/bughunteruniversity/nonvuln/phishing-with-window-opener)

So, our proposal would be to remove this addition in order to work as in previous versions (that's why we've created a "bug" since we did not find any documentation about it) or, if you want to maintain it we propose:

-  to only set "rel" options when target is "_blank", not in any case like now (in fact following the recommendation, and if you want to maintain "compatibility" with old versions of FF add "rel=noreferrer", if not just use "rel=noopener") and we would add if href points to an external link (to other domain than grafana's)
-  make it optional by config file like other security settings like x-xss-protection (maybe this is the best option, avoiding breaking things)

Thanks for your work and help!

**Environment**:
- Grafana version: 7.5.4
- Data source type & version: -
- OS Grafana is installed on: docker
- User OS & Browser: tested on update Chrome & Firefox
- Grafana plugins: -
- Others: -

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
